### ISSUE TITLE
Asset pipeline ignores files?

### ISSUE THREAD

**Issue by drallgood** [octopress/asset-pipeline#7]
Hi. 
I've got a very weird issue. Some of my JS files aren't included in the generated file.
I managed to track this down to the point that they aren't even included in the array returned by ```Octopress.site.static_files```
Other files in the same directory work fine.

Any ideas why that is?

**Comment by drallgood** [comment]
Update:
I was wrong thinking they weren't included in the static_files array. There's something weird going on.

If you modify the file like this:
``` ruby
      def add_js
      	Octopress.site.static_files.each do |f| 
          puts "pathJ0: '#{f.path}'"
         end
        Octopress.site.static_files.each do |f| 
          puts "pathJ: #{f.path}"
          if f.path =~ /\.js$/ 
            puts "   match"
            @js << Javascript.new(self, f)
            Octopress.site.static_files.delete(f) if combine_js
          else
          	puts "  NO match"
          end
        end
        Octopress.site.static_files.each do |f| 
          puts "pathJ3: '#{f.path}'"
        end
      end
```

you get:
```
pathJ0: '/Users/patrice/git/brendamour.net/source/javascripts/github.js'
pathJ0: '/Users/patrice/git/brendamour.net/source/javascripts/jquery.ad-gallery.js'
pathJ0: '/Users/patrice/git/brendamour.net/source/javascripts/jquery.fancybox.pack.js'
pathJ0: '/Users/patrice/git/brendamour.net/source/javascripts/jquery.lazyyoutube.js'
pathJ0: '/Users/patrice/git/brendamour.net/source/javascripts/modernizr.js'
pathJ0: '/Users/patrice/git/brendamour.net/source/javascripts/octopress.js'
pathJ0: '/Users/patrice/git/brendamour.net/source/javascripts/pinboard.js'
pathJ0: '/Users/patrice/git/brendamour.net/source/javascripts/twitter.js'
pathJ: /Users/patrice/git/brendamour.net/source/javascripts/github.js
   match
pathJ: /Users/patrice/git/brendamour.net/source/javascripts/jquery.fancybox.pack.js
   match
pathJ: /Users/patrice/git/brendamour.net/source/javascripts/modernizr.js
   match
pathJ: /Users/patrice/git/brendamour.net/source/javascripts/pinboard.js
   match
pathJ3: '/Users/patrice/git/brendamour.net/source/javascripts/jquery.ad-gallery.js'
pathJ3: '/Users/patrice/git/brendamour.net/source/javascripts/jquery.lazyyoutube.js'
pathJ3: '/Users/patrice/git/brendamour.net/source/javascripts/octopress.js'
pathJ3: '/Users/patrice/git/brendamour.net/source/javascripts/twitter.js'
```

I've got a feeling that this happens because you manipulate the array while iterating over it. Cloning it indeed fixes this issue.

**Comment by imathis** [comment]
Hmm. I'll have a look. Thanks for posting the issue!

**Comment by imathis** [comment]
Thanks so much for filing this issue. It should be fixed in 1.1.5 which I've just released.
